_layouts/default.html

```diff
@@ -0,0 +1,149 @@
+        <div class="p-navigation__banner">
+          <div class="p-navigation__logo">
+            <a class="p-navigation__link" href="/practices/">
+              <img class="p-navigation__image" src="https://assets.ubuntu.com/v1/60d9b81e-picto-canonical.svg" alt="Canonical">
+              <h4 style="padding: 0.5rem 1rem; margin-bottom: 0">Web and design practices</h4>
+            </a>
+          </div>
+          <a href="#navigation" class="p-navigation__toggle--open" title="menu">Menu</a>
+          <a href="#navigation-closed" class="p-navigation__toggle--close" title="close menu">Close menu</a>
+        </div>
+        <nav class="p-navigation__nav">
+          <span class="u-off-screen">
+            <a href="#main-content">Jump to main content</a>
+          </span>
+          <ul class="p-navigation__links">
+            <li class="p-navigation__link"><a href="https://github.com/canonical-webteam/practices" class="p-link--external">GitHub repo</a></li>
+            <li class="p-navigation__link"><a href="https://github.com/canonical-webteam/practices/blob/master/CONTRIBUTING.md" class="p-link--external">How to contribute</a></li>
+          </ul>
+        </nav>
+      </header>
+
+      <aside class="p-sidebar" id="navigation">
+        <div class="p-sidebar__banner u-hide--medium u-hide--large">
+          <i class="p-sidebar__toggle p-icon--menu"></i>
+        </div>
+        <div class="p-sidebar__content u-hide--small">
+          <nav class="p-sidebar-nav">
+            <ul class="p-sidebar-nav__list" id="docs-list-unsorted">
+              {% comment %}Get top-level categories{% endcomment %}
+              {% assign categories = "" | split: ',' %}
+              {% for node in site.pages %}
+                {% assign url_parts = node.url | split: '/' %}
+
+                {% if url_parts.size == 3 %}
+                  {% unless categories contains url_parts[1] %}
+                    {% assign categories = categories | push: url_parts[1] %}
+                  {% endunless %}
+                {% endif %}
+              {% endfor %}
+
+              {% comment %}Iterate through categories{% endcomment %}
+              {% assign sorted_categories = categories | sort %}
+              {% for category in sorted_categories %}
+                <li class="p-sidebar-nav__item">
+                  <strong>{{ category | replace: "-", " " | capitalize }}</strong>
+        
+                  <ul class="p-sidebar-nav__list" id="docs-list-unsorted">
+                    {% for node in site.pages %}
+                      {% assign url_parts = node.url | split: '/' %}
+                      {% if url_parts[1] == category %}
+                        {% unless node.redirect.to %}
+                          <li class="p-sidebar-nav__item">
+                            <a class="p-link--soft" href="/practices{{ node.url }}">
+                              {{ url_parts[2] | replace: ".html", "" | replace: "-", " " | capitalize }}
```

Comment:
I don't think that's the best idea. I'd rather keep the solution more generic...

Reply:
We should discuss in meatspace

Reply:
I'm not happy to add a special case for "HTML" here. I think we should merge this first (based on the existing logic of just uppercasing the first character from the filename), and then discuss how to improve it in a follow-up PR.

Reply:
I see here it's why we get UX > Ux, can we add this condition?

Reply:
We are constrained here to functionality that comes with GitHub pages, which means we really only have the `replace` filter as a way to try to achieve this.

This is a lot harder for UX than it would be for HTML, because while HTML is unlikely to appear in the middle of a word, the same can't be said for UX. I can do `replace: "-ux-", "-UX-"` but that's not going to catch UX at the start or the end of the string, but if I do `replace: "ux-", "UX-"` or `replace: "-ux", "-UX"` then `flux-` would become `flUX-` and `-uxbridge` would become `-UXbridge`. I can't think of any easy way around this...